Valide password 리뷰 부탁드립니다.

honeyl3ee opened the pull request:
#3 

멤버 생성 dto를 추가했습니다.
멤버 생성 dto에는 이메일 유효성검사와 요구사항의 패스워드 유효성 검사 항목 일부가 포함되어 있습니다.

member service에 join method는 Long 타입을 반환하도록 수정했습니다.
추후 생성된 멤버의 아이디를 반환하도록 하면 될 것 같습니다.

dto는 임의로 만들었는데 명명 방식에 대해서 논의가 필요합니다.

honeyl3ee committed:
Init: Add member controll dto

멤버 생성 컨트롤러 DTO를 추가함.
Github issue #3

honeyl3ee committed:
Feat: Change join parmeter to DTO

가입 서비스의 파라미터를 디티오로 변경함
Github issue #3

honeyl3ee committed:
Feat: Change return value of join method

join 번환값을 Long type으로 변경함
Github issue #3

honeyl3ee committed:
Init: Add member controller

createMember 컨트롤러를 생성함.
파라미터로 패스워드와 이메일 유효성 검증이 된 dto를 받음
Github issue #3

honeyl3ee committed:
Comment: Add comment to createMember method

멤버 생성 메서드를 설명하는 주석을 추가함
Github issue #3

honeyl3ee committed:
Comment: Add comment to CreateMemberDto

CreateMemberDto를 설명하는 주석을 추가함
Github issue #3

kshshkim commented on src/main/java/beforespring/socialfeed/member/service/MemberService.java:
반환되는 Long이 무슨 의미인지 명확하지 않습니다.

honeyl3ee replied:
주석으로 반환값에 대한 설명을 추가했습니다.

kshshkim commented on src/main/java/beforespring/socialfeed/member/domain/PasswordHasherImpl.java:
PasswordHasherImpl 이라는 이름이지만, 실제로 implement하고 있지 않습니다.

honeyl3ee replied:
아직 구현체를 작성하지 않아서 삭제했습니다.

kshshkim commented on src/main/java/beforespring/socialfeed/member/controller/MemberController.java:
requestId라는 변수 네이밍이 모호합니다. 생성된 Member 계정의 ID임이 드러나도록 하는 것이 바람직해 보입니다.

honeyl3ee replied:
수정했습니다.

kshshkim commented on src/main/java/beforespring/socialfeed/member/controller/dto/CreateMemberDto.java:
숫자 문자 특수문자 중 2가지 이상이라는 조건 때문에 정규식이 과하게 길어졌습니다. regex를 분리해서 메서드로 넣는 것이 좋아보여요. 생성자에 검증 메서드를 넣어서 생성 시점에 체크하도록 처리할 수 있습니다.

honeyl3ee replied:
수정했습니다.

kshshkim replied:
테스트시 오류가 납니다. "`passwdThatShou!dBe0kay`"는 주어진 조건인 "숫자, 문자, 특수문자 중 2가지 이상을 포함해야 하고, 3회 이상 연속되는 문자는 사용이 불가합니다" 를 만족하는 패스워드이지만, 검증 오류가 발생합니다.

honeyl3ee replied:
버그 수정했습니다. 
또한 조건 별로 다른 예외를 발생시켜 어느 부분에서 오류가 발생하는지 명확하게 했습니다.

honeyl3ee committed:
Refactor: rename requestId to memberId

createMember에서 join의 반환값 변수명을 명확하게 수정함.
Github issue #3

honeyl3ee committed:
Feat: Add password pattern validation method

비밀번호 패턴 검사 시 정규식이 길어서 가독성을 위해 패턴 검증 메서드를 따로 추가하여, 생성 메서드에서 검증하도록 함.
Github issue #3

honeyl3ee committed:
Commend: Add comment about join return value

join의 반환값에 대한 설명을 주석에 추가함.
Github issue #3

kshshkim committed:
Test: 패스워드 검증 테스트

honeyl3ee committed:
Fix: password pattern validation method

숫자, 문자, 특수문자 조합 정규식을 수정함.
연속된 동일한 문자를 확인하는 로직은 match가 아닌 find 메서드를 이용하여 비밀번호 일부만 해당되어도 true를 반환하도록 수정함.
Github issue #3

honeyl3ee committed:
Test: Change expectable exception message

dto에서 발생하는 에러가 변경됨에따라 에러 메세지를 수정함
Github issue #3

kshshkim commented on src/main/java/beforespring/socialfeed/member/service/MemberService.java:
서비스 계층 인터페이스에는 @Service 혹은 @Transactional 어노테이션을 붙이지 않는 것이 좋습니다.

honeyl3ee committed:
Refactor: remove annotation

인터페이스에 작성한 어노테이션을 삭제함
Github issue #3
